sim/midas/src/main/cc/simif_emul.cc

```diff
@@ -241,6 +145,12 @@ simif_emul_t::cpu_managed_axi4_write(size_t addr, char *data, size_t size) {
   return size;
 }
 
+simif_emul_t::FPGAManagedStreamIOImpl::FPGAManagedStreamIOImpl(
+    simif_emul_t &simif, const AXI4Config &config)
+    : simif(simif), cpu_mem(config) {
+  cpu_mem.init((1ULL << cpu_mem.get_config().addr_bits), 512);
```

Comment:
It would be good to move the comment: https://github.com/firesim/firesim/pull/1366/files#diff-aa2fcf9f9400da990595a6b66fe3037c6ad63db976821601589bf8487b7c364fL81 which describes this magic number here